tests/test_uniswapv2.py

```diff
@@ -0,0 +1,148 @@
+        self._router_contract = self._get_contract(self.web3, self.router_abi['abi'], self.router_address)
+        self._factory_contract = self._get_contract(self.web3, self.factory_abi['abi'], self.factory_address)
+
+        print(self.router_address, self.factory_address, __name__)
+
+        # self.dai_token = ERC20Token(web3=self.web3, address=deploy_contract(self.web3,abi 'DAIToken'))
+        # self.token_transfer_proxy_address = deploy_contract(self.web3, 'TokenTransferProxy')
+        # self.exchange = ZrxExchange.deploy(self.web3, self.zrx_token.address, self.token_transfer_proxy_address)
+        # TODO: deploy the other uniswap v2 contracts
+        # self.web3.eth.contract(abi=json.loads(pkg_resources.resource_string('pymaker.deployment', f'abi/TokenTransferProxy.abi')))(address=self.token_transfer_proxy_address.address).functions.addAuthorizedAddress(self.exchange.address.address).transact()
+
+        # self.dgx = DSToken.deploy(self.web3, 'DGX')
+        self.dai = DSToken.deploy(self.web3, 'DAI')
+
+        # Kovan token addresses
+        self.token_a = Token("DAI", Address("0x4f96fe3b7a6cf9725f59d353f723c1bdb64ca6aa"), 18)
+        self.token_b = Token("USDC", Address("0x198419c5c340e8de47ce4c0e4711a03664d42cb2"), 6)
+        self.uniswap = UniswapV2(self.web3, self.token_a, self.token_b)
+
+    def test_get_block(self):
+        assert isinstance(self.uniswap.get_block(), int)
+
+    def test_approval(self):
+        # given
+        assert self.dai.allowance_of(self.our_address, self.router_address) == Wad(0)
+
+        # when
+        # self._router_contract.approve([self.dai], directly())
+        approval_function = directly()
+        
+        approval_function(self.dai, self.router_address, 'IUniswapV2Router02')
+
+        # then
+        assert self.dai.allowance_of(self.our_address, self.router_address) > Wad(0)
+
+    def test_getting_balances(self):
+        # given
+        # self.dgx.mint(Wad(17 * 10**9)).transact()
+        self.uniswap.get_account_token_balance(self.token_a)\
+            # mint(Wad.from_number(17)).transact()
+
+        # when
+        # balances = self.uniswap.get_balances(self.pair)
+        # then
+        # assert balances[0] == Wad.from_number(17)
+        # assert balances[1] == Wad.from_number(17)
+
+    def test_add_liquidity_eth(self):
+
+        eth_pair_amounts = {
+            "amount_token_desired": self.web3.toWei(24, 'ether'),
+            "amount_token_min": self.web3.toWei(21, 'ether'),
+            "amount_eth_min": self.web3.toWei(.1, 'ether')
+        }
+
+        approval_function = directly()
+        approval_function(self.dai, self.router_address, 'IUniswapV2Router02')
+
+        time.sleep(20)
+        # when
+        add_liquidity = self.uniswap.add_liquidity_eth(eth_pair_amounts, self.dai.address).transact()
+
+        # then
+        assert add_liquidity.result == True
+
+        # when
```

Comment:
Test seems incomplete

Reply:
Added additional working tests. That being said, the tests currently only work with a local ganache instance. I am working through getting it working the docker-compose managed parity node.

Reply:
Updated tests